### DISCUSSION TITLE
Customize OpenTCS ModelEditor

### DISCUSSION THREAD

**Discussion by naiveZhou** [openTCS/opentcs#25] [General]
Hi,someone tried to customize the OpenTCS ModelEditor like this and i know it not use java language.
![image](https://user-images.githubusercontent.com/18117945/176334958-aa1540ac-25f8-4347-ba63-ee1f05dbf639.png)
As you know,I want to import coordinate file that created by the Laser Navigation Machine.I have converted to json file and now  Is it possible to make it.

**Comment by swltr** [comment]
It is not really clear what you are trying / required to do and what you might need help with. Can you rephrase and/or provide more details?

**Reply by naiveZhou** [reply]
Hi,I'm sorry I didn't express clearly.I want to draw point cloud area on the openTCS-ModelEditor, the coordinate data obtained by laser navigation.Now.I use the built-in point to load the area map and this is the rendering of the realization.
![image](https://user-images.githubusercontent.com/18117945/178168899-5a902031-5378-4244-a052-019349618e5e.png)
Can the built-in point support the implementation for point clouds?Or where can I start overriding the drawing method？
For example,Is it possible for me to draw such an area by Java Graphic on the  openTCS-ModelEditor.
https://zhuanlan.zhihu.com/p/428968514

**Reply by swltr** [reply]
The openTCS plant model only knows about a directed graph of points - discrete logical vehicle positions that may also contain information about the respective physical pose (i.e. coordinates and/or orientation angle). openTCS does not maintain point clouds by itself. (Theoretically, you could store point clouds in key/value properties of the plant model itself, but the GUI would not show them.)

**Comment by naiveZhou** [comment]
ok,i get it,thank you very much for your reply.



 
From: Stefan Walter
Date: 2022-07-15 18:35
To: openTCS/opentcs
CC: smallzhou; Author
Subject: Re: [openTCS/opentcs] Customize OpenTCS ModelEditor (Discussion #25)
The openTCS plant model only knows about a directed graph of points - discrete logical vehicle positions that may also contain information about the respective physical pose (i.e. coordinates and/or orientation angle). openTCS does not maintain point clouds by itself. (Theoretically, you could store point clouds in key/value properties of the plant model itself, but the GUI would not show them.)
